File: risk-openapi-java-demo/src/main/java/com/netease/risk/openapi/demo/nep/DoubtfulCheckDemo.java
package com.netease.risk.openapi.demo.nep;

import com.alibaba.fastjson.JSONObject;
import com.netease.risk.openapi.demo.util.HttpUtil;
import com.netease.risk.openapi.demo.util.SignatureUtils;

import java.util.HashMap;
import java.util.Map;

/**
 * 端游反外挂嫌疑在线检测
 * 本接口用于反外挂嫌疑数据在线检测。
 */
public class DoubtfulCheckDemo {
    // 产品ID，从【易盾官网-服务管理-已开通业务】页面获取
    private static final String APPID = "your_app_id";

    // 密钥，从【易盾官网-服务管理-已开通业务】页面获取
    private static final String APP_KEY = "your_app_key";

    // 接口URL
    private static final String API_URL = "http://open-yb.dun.163.com/api/open/v1/nep/doubtful/check";

    // 随机码
    private static final String nonce = "111";

    public static void main(String[] args) throws Exception {
        Map<String, Object> params = new HashMap<>();
        // 调用接口当前时间，单位毫秒
        Long timeStamp = System.currentTimeMillis();
        // 使用appKey签名的数据，校验权限
        String token = SignatureUtils.genSignature(APP_KEY, APPID, String.valueOf(timeStamp), nonce);
        params.put("appId", APPID);
        params.put("timestamp", timeStamp);
        params.put("nonce", nonce);
        params.put("token", token);
        // 嫌疑上报数据，游戏方需要从反外挂客户端SDK获取该数据
        params.put("mrData", "xxx");
        // 玩家ip
        params.put("ip", "yyy");
        // 角色id
        params.put("roleId", "yyy");
        // 角色名称
        params.put("roleName", "yyy");
        // 服务器
        params.put("roleServer", "yyy");
        // 额外信息，游戏方可以自己构建json结构，最大长度：2048
        params.put("extData", "zzz");

        String response = HttpUtil.sendHttpPost(API_URL, JSONObject.toJSONString(params));
        JSONObject jsonObject = JSONObject.parseObject(response, JSONObject.class);
        Integer code = jsonObject.getInteger("code");
        String msg = jsonObject.getString("msg");
        String data = jsonObject.getString("data");
        if (code == 200) {
            System.out.printf("请求成功, data=%s%n", data);
        } else {
            System.out.printf("请求失败, msg=%s%n", msg);
        }
    }
}

File: risk-openapi-python-demo/nep/check/DoubtfulCheckDemo.py
import hashlib
import time
import random
import urllib.request as urlrequest
import urllib.parse as urlparse
import json


class DoubtfulCheckDemo(object):
    """端游反外挂嫌疑在线检测"""

    API_URL = "http://open-yb.dun.163.com/api/open/v1/nep/doubtful/check"

    def __init__(self, app_id, app_key):
        """
        Args:
            app_id (str) 产品ID，从【易盾官网-服务管理-已开通业务】页面获取
            app_key (str) 密钥，从【易盾官网-服务管理-已开通业务】页面获取
        """
        self.app_id = app_id
        self.app_key = app_key

    def gen_signature(self, timestamp, nonce):
        params = {"appId": self.app_id, "timestamp": timestamp, "nonce": nonce}
        buff = ""
        for k in sorted(params.keys()):
            buff += str(k) + str(params[k])
        buff += self.app_key
        return hashlib.md5(buff.encode("utf8")).hexdigest()

    def check(self, params):
        """请求易盾接口
        Args:
            params (object) 请求参数
        Returns:
            请求结果，json格式
        """
        params["appId"] = self.app_id
        params["appKey"] = self.app_key
        params["timestamp"] = int(time.time() * 1000)
        params["nonce"] = int(random.random() * 100000000)
        params["token"] = self.gen_signature(params["timestamp"], params["nonce"])

        try:
            headers = {"Content-Type": 'application/json'}
            params = json.dumps(params)
            params = bytes(params, 'utf8')
            request = urlrequest.Request(url=self.API_URL, data=params, headers=headers)
            content = urlrequest.urlopen(request, timeout=1).read()
            return json.loads(content)
        except Exception as ex:
            print("调用API接口失败:", str(ex))


if __name__ == "__main__":
    """示例代码入口"""
    APP_ID = "your_app_id"
    APP_KEY = "your_app_key"
    api = DoubtfulCheckDemo(APP_ID, APP_KEY)

    params = {
        "mrData": "xxx",  # 嫌疑上报数据，游戏方需要从反外挂客户端SDK获取该数据
        "ip": "yyy",  # 玩家ip
        "roleId": "yyy",  # 角色id
        "roleName": "yyy",  # 角色名称
        "roleServer": "yyy",  # 服务器
        "extData": "zzz",  # 额外信息，游戏方可以自己构建json结构，最大长度：2048
    }

    ret = api.check(params)

    code: int = ret["code"]
    msg: str = ret["msg"]
    if code == 200:
        print("msg=%s, data=%s" % (msg, ret["data"]))
    else:
        print("ERROR: code=%s, msg=%s" % (ret["code"], ret["msg"]))
